app_flutter/lib/task_box.dart

```diff
@@ -8,32 +8,52 @@ import 'package:cocoon_service/protos.dart' show Task;
 
 /// Displays information from a [Task].
 ///
-/// Shows a black box for unknown messages.
+/// If [Task.status] is "In Progress", it will show as a "New" task
+/// with a [CircularProgressIndicator] in the box.
+/// Shows a black box for unknown statuses.
 class TaskBox extends StatelessWidget {
   const TaskBox({Key key, @required this.task}) : super(key: key);
 
   /// [Task] to show information from.
   final Task task;
 
-  /// A lookup table to define the background color for this ResultBox.
+  /// Status messages that map to TaskStatus enums.
+  /// TODO(chillers): Remove these and use TaskStatus enum when available. https://github.com/flutter/cocoon/issues/441
+  static const String statusFailed = 'Failed';
+  static const String statusNew = 'New';
+  static const String statusSkipped = 'Skipped';
+  static const String statusSucceeded = 'Succeeded';
+  static const String statusUnderperformed = 'Underperformed';
+  static const String statusInProgress = 'In Progress';
+
+  /// A lookup table to define the background color for this TaskBox.
   ///
-  /// The result messages are based on the messages the backend sends.
-  static const resultColor = <String, Color>{
-    'Failed': Colors.red,
-    'In Progress': Colors.purple, // v1 used the 'New' color while spinning
-    'New': Colors.blue,
-    'Skipped': Colors.transparent,
-    'Succeeded': Colors.green,
-    'Underperformed': Colors.orange,
+  /// The status messages are based on the messages the backend sends.
+  static const statusColor = <String, Color>{
+    statusFailed: Colors.red,
+    statusNew: Colors.blue,
+    statusInProgress: Colors.blue,
+    statusSkipped: Colors.transparent,
+    statusSucceeded: Colors.green,
+    statusUnderperformed: Colors.orange,
   };
 
   @override
   Widget build(BuildContext context) {
     return Container(
       margin: const EdgeInsets.all(1.0),
-      color: resultColor.containsKey(task.status)
-          ? resultColor[task.status]
+      color: statusColor.containsKey(task.status)
+          ? statusColor[task.status]
           : Colors.black,
+      child: (task.status == statusInProgress)
+          ? const Padding(
+              padding: const EdgeInsets.all(15.0),
```

Comment:
Since you're already in a const context here the `const` is redundant. I think there is a lint that will catch this?

```suggestion
              padding: EdgeInsets.all(15.0),
```

Reply:
Flutter format didn't change anything, and flutter analyze doesn't throw errors. Is there another tool for stuff like this?

Reply:
Check out https://dart.dev/guides/language/analysis-options

Reply:
Should this project use the same analysis_options.yaml that the Dart backend project is using?

Reply:
They should probably be consistent